src/mon/MonmapMonitor.cc

```diff
@@ -1298,13 +1298,17 @@ void MonmapMonitor::trigger_degraded_stretch_mode(const set<string>& dead_mons)
 {
   dout(20) << __func__ << dendl;
   pending_map.stretch_marked_down_mons.insert(dead_mons.begin(), dead_mons.end());
+  pending_map.disallowed_leaders.insert(dead_mons.begin(), dead_mons.end());
   propose_pending();
 }
 
 void MonmapMonitor::trigger_healthy_stretch_mode()
 {
   dout(20) << __func__ << dendl;
   pending_map.stretch_marked_down_mons.clear();
+  pending_map.disallowed_leaders.clear();
+  const string& tiebreaker_mon = pending_map.tiebreaker_mon;
+  pending_map.disallowed_leaders.insert(tiebreaker_mon);
```

Comment:
There's one particular problem with manipulating the monmap's ```disallowed_leaders``` set here:
users can specify disallowed_leaders manually.

That's why this set is distinguished from what we pass in to the Elector's disallowed_leaders; ```Monmap::disallowed_leaders``` is meant to contain admin-specified blocks, the ```stretch_marked_down_mons``` specifies stretch rule blocks, and the ```tiebreaker_mon``` is the tiebreaker and so also not allowed to be a leader. We need to give Elector the union of those sets, and we definitely don't want to do have a situation where we have site1 and site2 with mons site1-1,site1-2,site2-1,site2-2, an admin blocks site1-1, then site1 dies and comes back and abruptly site1-1 becomes the leader despite having been manually blocked.

Reply:
Yep that make sense, you're right this would definitely cause a problem in this scenario.